[Microsoft.ResourceHealth] Add metadata api

mudit794 opened the pull request:
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

mudit794 committed:
Update ResourceHealth.json

mudit794 committed:
Add files via upload

mudit794 committed:
Update GetMetadataEntity.json

mudit794 committed:
Update ListMetadata.json

mudit794 committed:
Update ResourceHealth.json

mudit794 committed:
Update ResourceHealth.json

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-js

azuresdkci commented:
Can one of the admins verify this patch?

AutorestCI commented:
# Automation for azure-sdk-for-java

AutorestCI commented:
# Automation for azure-sdk-for-go

mudit794 committed:
Update Events_ListBySubscriptionId.json

mudit794 committed:
Update Events_ListBySingleResource.json

mudit794 committed:
Rename GetMetadataEntity.json to Metadata_GetMetadataEntity.json

mudit794 committed:
Rename ListMetadata.json to Metadata_ListMetadata.json

mudit794 committed:
Update ResourceHealth.json

mudit794 committed:
Update ResourceHealth.json

mudit794 committed:
Update Metadata_ListMetadata.json

mudit794 committed:
Update Metadata_ListMetadata.json

veronicagg commented on specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2018-08-01/ResourceHealth.json:
```diff
@@ -1089,6 +1161,84 @@
+        }
+      }
+    },
+    "MetadataEntity": {
```
is this an ARM resource? should you have something similar to [what you have here](https://github.com/Azure/azure-rest-api-specs/blob/affdae4a844ea5e0ced7ea805a9d9007e625f012/specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2018-08-01/ResourceHealth.json#L661).

veronicagg reviewed (changes requested):
Left some comments inline. 
Also your swagger file is not referenced from the readme.md file https://github.com/Azure/azure-rest-api-specs/blob/affdae4a844ea5e0ced7ea805a9d9007e625f012/specification/resourcehealth/resource-manager/readme.md 
Should you have your latest package tag pointing to this specification? The readme.md file is the configuration file for AutoRest so SDKs can be regenerated for your service. Feel free to let me know if you have questions on this.

mudit794 committed:
Update ResourceHealth.json

mudit794 committed:
Update readme.go.md

mudit794 committed:
Update readme.md

mudit794 committed:
Update readme.md

mudit794 committed:
Update readme.go.md

mudit794 committed:
Update readme.md

mudit794 commented:
> Left some comments inline.
> Also your swagger file is not referenced from the readme.md file https://github.com/Azure/azure-rest-api-specs/blob/affdae4a844ea5e0ced7ea805a9d9007e625f012/specification/resourcehealth/resource-manager/readme.md
> Should you have your latest package tag pointing to this specification? The readme.md file is the configuration file for AutoRest so SDKs can be regenerated for your service. Feel free to let me know if you have questions on this.







> Left some comments inline.
> Also your swagger file is not referenced from the readme.md file https://github.com/Azure/azure-rest-api-specs/blob/affdae4a844ea5e0ced7ea805a9d9007e625f012/specification/resourcehealth/resource-manager/readme.md
> Should you have your latest package tag pointing to this specification? The readme.md file is the configuration file for AutoRest so SDKs can be regenerated for your service. Feel free to let me know if you have questions on this.

Fixed it.

veronicagg commented on specification/resourcehealth/resource-manager/readme.md:
````diff
@@ -49,6 +49,15 @@ input-file:
 - Microsoft.ResourceHealth/stable/2015-01-01/resourcehealth.json
 ```
 
+### Tag: package-2018-08-preview
````
should this tag be your default tag in line 29? or do you want to keep package-2017-07 as the default? The default tag is what's used to generate SDKs by default, unless you particularly specify and let us know something else.

mudit794 replied:
I am adding this new change to preview api. We will update api version to 2018-08-01 once it goes GA (basically not preview anymore). So will keep 2017-07 as default for now,.

veronicagg reviewed (changes requested):
Thanks for addressing the comments, I have one more question inline. 
Additionally, I noticed you're configured to generate 
swagger-to-sdk:
  - repo: azure-sdk-for-go
  - repo: azure-sdk-for-js
  - repo: azure-sdk-for-node

though not java, python, ruby. Fee free to add to readme.md file for generation.

mudit794 committed:
Update readme.md

mudit794 committed:
Update readme.md

mudit794 commented:
> Thanks for addressing the comments, I have one more question inline.
> Additionally, I noticed you're configured to generate
> swagger-to-sdk:
> 
> * repo: azure-sdk-for-go
> * repo: azure-sdk-for-js
> * repo: azure-sdk-for-node
> 
> though not java, python, ruby. Fee free to add to readme.md file for generation.

Done.

veronicagg reviewed (approved):
Thanks for addressing the comments @mudit794 !
